src/components/FormElements/ResourceFeePickerWithQuery.tsx

```diff
@@ -0,0 +1,185 @@
+    // If the user has fetched the min resource fee, update the input's value
+    if (simulateTxData?.result?.minResourceFee) {
+      const syntheticEvent = {
+        target: {
+          value: simulateTxData.result.minResourceFee.toString(),
+        },
+      } as React.ChangeEvent<HTMLInputElement>;
+
+      onChange(syntheticEvent);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [simulateTxData?.result?.minResourceFee]);
+
+  // Create a sample transaction to simulate to get the min resource fee
+  const buildTxToSimulate = () => {
+    try {
+      // We add a bogus fee to simualte to fetch the min resource fee from the RPC
+      const BOGUS_RESOURCE_FEE = "100";
+
+      let builtXdr, contractDataXDR;
+
+      try {
+        contractDataXDR = getContractDataXDR({
+          contractAddress: operation.params.contract,
+          dataKey: operation.params.key_xdr,
+          durability: operation.params.durability,
+        });
+      } catch (e) {
+        throw new Error(`Failed to generate contract data XDR: ${e}`);
+      }
+
+      if (!contractDataXDR) {
+        throw new Error("Failed to fetch contract data XDR");
+      }
+
+      const sorobanData = getSorobanTxData({
+        contractDataXDR,
+        operationType: operation.operation_type as SorobanOpType,
+        fee: BOGUS_RESOURCE_FEE, // simulate purpose only
+      });
+
+      if (sorobanData) {
+        builtXdr = buildSorobanTx({
+          sorobanData,
+          params: txnParams,
+          sorobanOp: {
+            ...operation,
+            params: {
+              ...operation.params,
+              resource_fee: BOGUS_RESOURCE_FEE,
+            },
+          },
+          networkPassphrase: network.passphrase,
+        }).toXDR();
+      } else {
+        throw new Error("Failed to build Soroban transaction data");
+      }
+      return builtXdr;
+    } catch (e) {
+      setErrorMessage(`${e}`);
+      return;
+    }
+  };
+
+  return (
+    <PositiveIntPicker
+      key={id}
+      id={id}
+      label={label}
+      placeholder={placeholder}
+      labelSuffix={labelSuffix}
+      value={value}
+      error={error || errorMessage}
+      onChange={(e) => {
+        onChange(e);
+        setErrorMessage(undefined);
+      }}
+      infoLink={infoLink}
+      note={note}
+      disabled={disabled}
+      rightElement={
+        <InputSideElement
+          variant="button"
+          onClick={async () => {
+            const sampleTxnXdr = buildTxToSimulate();
+
+            if (!sampleTxnXdr) {
+              return;
+            }
+
+            await simulateTx({
+              rpcUrl: network.rpcUrl,
+              transactionXdr: sampleTxnXdr,
+              headers: getNetworkHeaders(network, "rpc"),
+            });
+          }}
+          placement="right"
+          // if we can't build a txn to simulate, we can't fetch the min resource fee
+          disabled={!(isValid.params && isOperationValidForSimulateTx)}
```

Comment:
@quietbits disabling the button until all operation params (except the resource fee) is filled